### ISSUE TITLE
<link> is not working for rich text formatting

### ISSUE THREAD

**Issue by ssdh233** [formatjs/react-intl#1464]
First of all, thanks for maintaining this great package!

This example from [documentation](https://github.com/formatjs/react-intl/blob/master/docs/Components.md#rich-text-formatting) is not working as expected.
```javascript
<FormattedMessage
  id="test-link"
  defaultMessage="To buy a shoe, <link>visit our website</link> and <cta>buy a shoe</cta>"
  values={{
    link: msg => (
      <a
        class="external_link"
        target="_blank"
        href="https://www.shoe.com/"
      >
        {msg}
      </a>
    ),
    cta: msg => <strong class="important">{msg}</strong>
  }}
/>
```

### Current behavior

Current output is plain string
```
To buy a shoe, <link>visit our website</link> and <cta>buy a shoe</cta>
```

### Expected behavior

`<link>` should be rendered as a html `<a>` tag and `<cta>` should be rendered as a html `<strong>` tag. So it should be something like

To buy a shoe, <a href="">visit our website</a> and <strong>buy a shoe</strong>.

Or if there is any reason that we cannot use `<link>` for rich text formatting, [the example here](https://github.com/formatjs/react-intl/blob/master/docs/Components.md#rich-text-formatting) should be changed and it would be better if the reason could be documented.

### Step to reproduce for BUG REPORT

I made an example on code sandbox, please have a look :)
https://codesandbox.io/embed/react-intl-bug-report-link-rich-text-2rpfp

### Your Environment

|          Executable | Version |
| ------------------: | :------ |
| `npm ls react-intl` | 3.1.13 |
|      `npm ls react` | 16.9.0 |

| OS   | Version |
| ---- | ------- |
| macOS | 10.12.6 |

| Browser | Version |
| ------- | ------- |
| Chrome | 61.0.3163.71 |

**Comment by longlho** [comment]
Yes link is a void element so the example is wrong. I’ll fix it soon

**Comment by longlho** [comment]
fixed in a090cc1f0bb9d0b3da3060b87c67671f08ecac9d
